terminus-ui/expansion-panel/src/trigger/expansion-panel-trigger.component.ts

```diff
@@ -0,0 +1,218 @@
+import {
+  FocusableOption,
+  FocusMonitor,
+  FocusOrigin,
+} from '@angular/cdk/a11y';
+import { hasModifierKey } from '@angular/cdk/keycodes';
+import {
+  ENTER,
+  SPACE,
+} from '@terminus/ngx-tools/keycodes';
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  Host,
+  Inject,
+  Input,
+  OnDestroy,
+  Optional,
+  ViewEncapsulation,
+} from '@angular/core';
+import {
+  EMPTY,
+  merge,
+  Subscription,
+} from 'rxjs';
+import { filter } from 'rxjs/operators';
+import { untilComponentDestroyed } from '@terminus/ngx-tools';
+
+import { tsExpansionPanelAnimations } from './../expansion-animations';
+import {
+  TS_EXPANSION_PANEL_DEFAULT_OPTIONS,
+  TsExpansionPanelComponent,
+  TsExpansionPanelDefaultOptions,
+} from './../expansion-panel.component';
+
+
+/**
+ * Trigger to open/close a {@link TsExpansionPanelComponent}
+ *
+ * @example
+ * <ts-expansion-panel>
+ *               <ts-expansion-panel-trigger
+ *                 collapsedHeight="100px"
+ *                 expandedHeight="150px"
+ *               >
+ *                 Panel trigger
+ *               </ts-expansion-panel-trigger>
+ *
+ *               Panel content
+ * </ts-expansion-panel>
+ *
+ * <example-url>https://getterminus.github.io/ui-demos-master/components/expansion-panel</example-url>
+ */
+@Component({
+  animations: [
+    tsExpansionPanelAnimations.indicatorRotate,
+    tsExpansionPanelAnimations.expansionTriggerHeight,
+  ],
+  selector: 'ts-expansion-panel-trigger',
+  styleUrls: ['./expansion-panel-trigger.component.scss'],
+  templateUrl: './expansion-panel-trigger.component.html',
+  host: {
+    class: 'ts-expansion-panel__trigger',
+    role: 'button',
+    '[attr.id]': 'panel.triggerId',
+    '[attr.tabindex]': 'disabled ? -1 : 0',
+    '[attr.aria-controls]': 'panel.id',
+    '[attr.aria-expanded]': 'isExpanded',
+    '[attr.aria-disabled]': 'panel.disabled',
+    '[class.ts-expansion-panel__trigger--expanded]': 'isExpanded',
+    '(click)': 'toggle()',
+    '(keydown)': 'keydown($event)',
+    '[@expansionHeight]': `{
+        value: currentPanelExpandedState,
+        params: {
+          collapsedHeight: collapsedHeight,
+          expandedHeight: expandedHeight
+        }
+    }`,
+  },
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TsExpansionPanelTriggerComponent implements OnDestroy, FocusableOption {
+  /**
+   * Determine the current expanded state string of the panel
+   */
+  get currentPanelExpandedState(): string {
+    return this.panel.currentExpandedState;
+  }
+
+  /**
+   * Determine if the panel is currently expanded
+   */
+  get isExpanded(): boolean {
+    return this.panel.expanded;
+  }
+
+  /**
+   * Whether the associated panel is disabled.
+   *
+   * Implemented as a part of `FocusableOption`.
+   */
+  get disabled(): boolean {
+    return this.panel.disabled;
```

Comment:
is this the native disabled/different from our isDisabled?

Reply:
The CKD implements the input as `disabled` but our library convention is a bit more explicit as `isDisabled`. So I've created inputs that just set the value on the existing input. That way consumers use our API but the code itself continues using what the CDK provides.